src/Compiler.php

```diff
@@ -5336,13 +5336,124 @@ public function findImport($url)
             }
         }
 
-        if ($urls) {
-            if (! $hasExtension || preg_match('/[.]scss$/', $url)) {
-                throw $this->error("`$url` file not found for @import");
-            }
+        throw $this->error("`$url` file not found for @import");
+    }
+
+    private function resolveImportPath($url, $baseDir)
+    {
+        $path = rtrim($baseDir, '/').'/'.ltrim($url, '/');
+
+        $hasExtension = preg_match('/.scss$/', $url);
+
+        if ($hasExtension) {
+            return $this->checkImportPathConflicts($this->tryImportPath($path));
         }
 
-        return null;
+        $result = $this->checkImportPathConflicts($this->tryImportPathWithExtensions($path));
+
+        if (!\is_null($result)) {
+            return $result;
+        }
+
+        return $this->tryImportPathAsDirectory($path);
+    }
+
+    /**
+     * @param string[] $paths
+     *
+     * @return string|null
+     */
+    private function checkImportPathConflicts(array $paths)
+    {
+        if (\count($paths) === 0) {
+            return null;
+        }
+
+        if (\count($paths) === 1) {
+            return $paths[0];
+        }
+
+        $formattedPrettyPaths = [];
+
+        foreach ($paths as $path) {
+            $formattedPrettyPaths[] = '  ' . $this->getPrettyPath($path);
+        }
+
+        throw $this->error("It's not clear which file to import. Found:\n" . implode("\n", $formattedPrettyPaths));
+    }
+
+    /**
+     * @param string $path
+     *
+     * @return string[]
+     */
+    private function tryImportPathWithExtensions($path)
+    {
+        $result = $this->tryImportPath($path.'.scss');
+
+        if ($result) {
+            return $result;
+        }
+
+        return $this->tryImportPath($path.'.css');
+    }
+
+    /**
+     * @param string $path
+     *
+     * @return string[]
+     */
+    private function tryImportPath($path)
+    {
+        $partial = dirname($path).'/_'.basename($path);
```

Comment:
Is it the expected behaviour that `@import '_partial';`  will try to import  `__partial.scss`?

I had the feeling that an explicit partial reference in the `@import` should just be taken as is, not trying to partialize the already partial file (at least that's what scssphp was doing before).
Am I wrong?

Reply:
dart-sass will do it too.

Reply:
yep seems more a flaw in the design than a real feature :p 
but if dart-saas do that, let's stick on it

This seems all good for me besides that.

Reply:
I opened https://github.com/sass/sass/issues/2940 to clarify that.

Reply:
@Cerdic it's `dart-sass`, not `dart-saas` :smile:

Reply:
The Sass team answered that this behavior is the same in dart-sass and ruby-sass, and not worth a BC break to undo it, even though they would have preferred doing that differently retrospectively (that was probably not intended). So I think it is good we follow the same rule.